### DISCUSSION TITLE
Normalize output in regression/forecasting

### DISCUSSION THREAD

**Discussion by vrodriguezf** [timeseriesAI/tsai#257] [Q&A] [answered]
Hi,

Is there anything like a batch_tfm for the target `y` when it is a number, like in a regression, or a tensor, like in a multi-horizon (or multi-step) forecasting problem? I don't even know if that's something that fastai contemplates

Thanks!

**Comment by oguiza** [comment, marked as answer]
It's not possible out of the box. And I'm not sure there's any benefit in doing that. Maybe for very large outputs. I don't know. 
In any case, if you want to create a reversible transform, you'd need to create a new tensor subtype (for example, LabelTensor) and a new transform specific for that tensor subtype, with a encode and decode method (similar to TSStandardize).

**Comment by oguiza** [comment]
I'll move this to Discussions as it's more a question than an issue.

**Comment by yangtzech** [comment]
Hi, @oguiza . I am also faced with this kind of issue. I am currently dealing with a dataset with output in small values for a regression problem. The RMSE and MAE are low and MAPE is high. So it is possible to scale the small value to a larger range, e.g. from [0, 1e-3] to [0, 1]?

**Reply by vrodriguezf** [reply]
See the answer to this thread, subtyping the y tensor and defining encode/decode methods should be the way to go

**Reply by yangtzech** [reply]
> 

Thanks. I see and will give it a go.

**Reply by yangtzech** [reply]
I checked the source code of TSStandardize. The encode/decode method scales the x value like sklearn StandardScaler. But it seems kind of complicated to add TSStandardizeForY. And I am also kind of confused about what 'subtyping the y tensor' means.

**Reply by oguiza** [reply]
>  I am currently dealing with a dataset with output in small values for a regression problem. The RMSE and MAE are low and MAPE is high. So is it possible to scale the small value to a larger range, e.g. from [0, 1e-3] to [0, 1]?

Hi @yangtzech,
I'm not sure that normalization would actually work. In my experience (and it may not be applicable to your case) I have not seen performance improvements when scaling the target variable. While training, the model will always try to minimize the loss, regardless of its value. 
What I always try to do first is try to get better than the expected result in the training set (you can set train_metrics to True in the ts_learner). In other words, overfit the training set to ensure the model you are using is capable of learning the task at hand.
If you still want to scale the target, I'd suggest that the easiest thing to do is to normalize it before passing the y to the dataloaders.

**Reply by yangtzech** [reply]
> 

Thanks! I will try.
